packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -51,11 +53,41 @@ function useBlockEditorSettings( settings, hasTemplate ) {
 				canUser( 'create', 'media' ),
 				true
 			),
+			baseUrl: getUnstableBase()?.url || '',
 		};
 	}, [] );
 
 	const { undo } = useDispatch( editorStore );
 
+	// Temporary home - should this live in `core-data`?
+	function fetchRichUrlData( url, fetchOptions = {} ) {
+		const emptyDataSet = {};
+
+		const fetchingBaseUrl = '' === baseUrl;
+
+		// If the baseUrl is still resolving then return
+		// empty data for this request.
+		if ( fetchingBaseUrl ) {
+			return Promise.resolve( emptyDataSet );
+		}
```

Comment:
When will this happen? Should we instead wait for it to be resolved? So instead of using `useSelect`, we could probably just use `subscribe` and wrap it inside a promise.

Reply:
The state isn't populated until the selector resolves. Therefore there is the possibility that we're comparing against nothing.

Therefore I thought I'd set the default to an empty string so that we can tell the difference between:

1. This selector isn't yet populated (empty string `''`).
2. This selector will never be populated (`undefined`)- which would presumably be the case if you're in a non-WP env.

How would the subscribe look? Not sure I've seen this pattern.

Reply:
`useSelect` is actually just a wrapper of `subscribe` (well, not actually, but a wrapper of a more advanced version of subscribe). So it would look something like this.

```js
import { subscribe, select } from '@wordpress/data';

const baseUrlPromise = new Promise( ( resolve ) => {
	const unsubscribe = subscribe( () => {
		const baseUrl = select( coreStore ).getUnstableBase()?.url;
		if ( baseUrl ) {
			resolve( baseUrl );
			unsubscribe();
		}
	} );
} );
```

I understand if it's too much work though. Where does this `getUnstableBase()` come from? Why isn't it populated with initial data? I can't seem to find the definition of this method 🤔.

Reply:
> Where does this getUnstableBase() come from? Why isn't it populated with initial data? I can't seem to find the definition of this method 🤔.

Same! I just tried searching for it and it's nowhere in the source. Does it come from WP Core?

Reply:
@gwwar used it in the `Home` block code which is where I found it. Do you know where this is from Kerry?

Reply:
Ok I found some info here https://github.com/WordPress/gutenberg/pull/30812#issuecomment-819748485

Reply:
I've now used the resolver to determine whether the baseUrl is available.